crates/rslint_parser/src/syntax/jsx/mod.rs

```diff
@@ -583,6 +585,31 @@ fn parse_jsx_expression_block(p: &mut Parser, kind: ExpressionBlock) -> ParsedSy
 
     p.bump(T!['{']);
 
+    // test jsx jsx_children_spread
+    // <div>{...a}</div>
+    if kind == ExpressionBlock::Children && p.at(T![...]) {
+        p.expect(T![...]);
+        let expr = super::expr::parse_expression(p, ExpressionContext::default()).map(|mut m| {
+            match m.kind() {
+                JsSyntaxKind::IMPORT_META
+                | JsSyntaxKind::NEW_TARGET
+                | JsSyntaxKind::JS_CLASS_EXPRESSION
+                | JsSyntaxKind::JS_SEQUENCE_EXPRESSION => {
+                    let err = p
+                        .err_builder("This expression is not valid as a JSX spread expression")
+                        .primary(m.range(p), "");
+                    p.error(err);
+                    m.change_to_unknown(p);
+                    m
+                }
+                _ => m,
+            }
+        });
+        expr.or_add_diagnostic(p, expected_expression);
+        p.expect(T!['}']);
+        return Present(m.complete(p, JSX_SPREAD_CHILD));
```

Comment:
I would recommend merging the code with the "regular" expression parsing to make sure things like `bump_with_context` are correctly handled

```suggestion
        let kind = match kind {
             ExpressionBlock::Attribute => JsSyntaxKind::JSX_EXPRESSION_ATTRIBUTE_VALUE,
             ExpressionBlock::Children => {
                  if p.eat(T![...]) {
                       JSX_SPREAD_CHILD
                  } else {
                      JSX_EXPRESSION_CHILD
                  }
             }
        };
```